Source/WebCore/rendering/MotionPath.cpp

```diff
@@ -194,19 +194,36 @@ std::optional<Path> MotionPath::computePathForRay(const RayPathOperation& rayPat
     return path;
 }
 
+static FloatRoundedRect offsetRectForData(const MotionPathData& data)
+{
+    auto rect = data.containingBlockBoundingRect;
+    auto shiftedPoint = data.offsetFromContainingBlock;
+    shiftedPoint.scale(-1);
+    rect.setLocation(shiftedPoint);
+    return rect;
+}
+
 std::optional<Path> MotionPath::computePathForBox(const BoxPathOperation&, const TransformOperationData& data)
 {
     if (auto motionPathData = data.motionPathData()) {
         Path path;
-        auto rect = motionPathData->containingBlockBoundingRect;
-        auto shiftedPoint = motionPathData->offsetFromContainingBlock;
-        shiftedPoint.scale(-1);
-        rect.setLocation(shiftedPoint);
-        path.addRoundedRect(rect, PathRoundedRect::Strategy::PreferBezier);
+        path.addRoundedRect(offsetRectForData(*motionPathData), PathRoundedRect::Strategy::PreferBezier);
         return path;
     }
     return std::nullopt;
 }
 
 
+std::optional<Path> MotionPath::computePathForShape(const ShapePathOperation& pathOperation, const TransformOperationData& data)
+{
+    if (auto motionPathData = data.motionPathData()) {
+        auto& shape = pathOperation.basicShape();
+        auto rect = motionPathData->containingBlockBoundingRect.rect();
+        if (is<BasicShapeInset>(shape))
+            rect = offsetRectForData(*motionPathData).rect();
```

Comment:
I don't think it's just inset(), it's all the shapes that should be unaffected by properties adding more offsets from the container. Sadly we don't have testcases exercising this bug on other shapes

Reply:
Right. Switched to all shapes except for path.